src/analyzer/protocol/ssl/SSL.cc

```diff
@@ -310,28 +314,28 @@ bool SSL_Analyzer::TryDecryptApplicationData(int len, const u_char* data, bool i
         else
             EVP_DecryptInit(ctx, EVP_aes_256_gcm(), s_wk, s_aead_nonce.data());
 
-        EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_GCM_SET_TAG, 16, encrypted + encrypted_len);
+        EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_GCM_SET_TAG, 16, const_cast<u_char*>(encrypted + encrypted_len));
 
         // AEAD tag
-        std::basic_string<unsigned char> s_aead_tag;
+        byte_buffer s_aead_tag;
         if ( is_orig )
             s_aead_tag = fmt_seq(c_seq);
         else
             s_aead_tag = fmt_seq(s_seq);
 
+        assert(s_aead_tag.size() == 13);
```

Comment:
Is there a reason for the move of this assert?

Just looking at the code, it seems like it might always trigger after this change.

Reply:
if size wasn't at least 13, accessing `s_aead_tag[12]` would be invalid, so seemed reasonable to pre-assert before potentially accessing out-of-bounds.

Reply:
I think it won't because it's reserved in `fmt_seq` which is always called https://github.com/zeek/zeek/blob/7b3bc891f4b4e132b8ec37b9b2d1b7f5235b2769/src/analyzer/protocol/ssl/SSL.cc#L39

@awelzel can confirm since that's from the [suggested commit](https://github.com/zeek/zeek/commit/b04164b78f38eab03b35cf873cced49435e811e1), but that was my understanding